File: ga/glm4_result_2024_6/result/1968B_glm/8_1968B.c
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let t: i32 = input.trim().parse().expect("Please type a number");

    for _ in 0..t {
        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Failed to read line");
        let mut parts = input.trim().split_whitespace();
        let n: usize = parts.next().expect("Please provide a number").parse().expect("Please type a number");
        let m: usize = parts.next().expect("Please provide a number").parse().expect("Please type a number");
        let sn: String = parts.next().expect("Please provide a string").to_string();
        let sm: String = parts.next().expect("Please provide a string").to_string();

        let mut i = 0;
        let mut j = 0;

        while i < n && j < m {
            if sn.chars().nth(i) == sm.chars().nth(j) {
                i += 1;
            }
            j += 1;
        }

        println!("{}", i);
    }
}

File: ga/gpt3_5_result_2024_6/result/1968B_cpp_gpt/8_1968B.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines().map(|line| line.unwrap());
    
    let t: i32 = lines.next().unwrap().parse().unwrap();
    
    for _ in 0..t {
        let input: Vec<i32> = lines.next().unwrap().split_whitespace()
            .map(|x| x.parse().unwrap()).collect();
        let x = input[0];
        let k = input[1];
        
        let a: String = lines.next().unwrap();
        let b: String = lines.next().unwrap();
        
        let mut ans = 0;
        for i in b.chars() {
            if ans < a.len() && a.chars().nth(ans).unwrap() == i {
                ans += 1;
            }
        }
        
        println!("{}", ans);
    }
}